lib/publiccloud/instance.pm

```diff
@@ -346,59 +348,74 @@ sub wait_for_ssh {
+    # ssh OK or proceed_on_failure
+    my $duration = time() - $start_time;
+    $status = (defined($status) and $status == 0) ? 1 : 0;    # shell-code to perl-test convertion
+    $status = 0 unless $args{timeout} > $duration;
+    # On boottime test we do hard reboot which may change the instance address:
+    script_run("ssh-keyscan $args{public_ip} | tee -a ~/.ssh/known_hosts")
+      if ($status and (get_var('PUBLIC_CLOUD_PERF_COLLECT') or get_var('PUBLIC_CLOUD_CHECK_BOOT_TIME')));
+
+    my ($sshout, $sysout);
+    if ($status) {
+        $sshout = "SSH port is open.\n";
     }
-
-    script_run("ssh  -i /root/.ssh/id_rsa $args{username}\@$args{public_ip} true", timeout => 360);
-    # Debug output: We have occasional error in 'journalctl -b' - see poo#96464 - this will be removed soon.
-    # Exclude 'mr_test/saptune' test case as it will introduce random softreboot failures.
-    if (!get_var('PUBLIC_CLOUD_SLES4SAP')) {
-        $self->run_ssh_command(cmd => 'sudo journalctl -b', proceed_on_failure => 1, username => $args{username}, timeout => 360);
+    else {    # proceed_on_failure
+        $sshout = "SSH port is not open failed access to $instance_msg\n";
     }
-
-    unless ($args{proceed_on_failure}) {
-        my $error_msg;
-        if ($check_port) {
-            $error_msg = sprintf("Unable to reach SSH port of instance %s with public IP:%s within %d seconds", $self->{instance_id}, $self->{public_ip},
-                $args{timeout});
+    # check also remote system is running, looping by --wait option:
+    if ($args{systemup_check} and $status) {
+        # timeout recalculated removing consumed time until now
+        $sysout = $self->ssh_script_output(cmd => 'sudo systemctl is-system-running --wait',
+            timeout => $args{timeout} - $duration, proceed_on_failure => 1, username => $args{username});
+
+        $duration = time() - $start_time;    # time update after check result
+        if ($sysout =~ m/running/) {    # startup OK
+            $status = 1;
+            $sysout .= "\nSystem successfully booted: $instance_msg";
+        }
+        elsif ($sysout =~ m/degraded/) {    # OK with failed services: collection
+            $status = 1;
+            $sysout .= "\n" . $self->ssh_script_output(cmd => 'sudo systemctl --failed', proceed_on_failure => 1, username => $args{username});
+        }
+        else {    # FAIL: $sysout ~ maintenance|stopping|offline|unknown or no pubkey or else
+            $status = 0;
+            $sysout .= "\nCan not reach systemd target on $instance_msg";
         }
-        else {
-            $error_msg = sprintf("Can not reach systemd target on instance %s with public IP:%s within %d seconds",
-                $self->{instance_id}, $self->{public_ip}, $args{timeout});
+        # Log upload
+        if (!get_var('PUBLIC_CLOUD_SLES4SAP') and $status > 0) {
```

Comment:
`$status` is being treated as a boolean elsewhere. I'd go here for readability with:
```
if (!get_var('PUBLIC_CLOUD_SLES4SAP') and $status) {
```

Reply:
Yes thanks, it is a typo due to a previous plan to use values < 0, then abandoned.